CARDS-2587: Malfunction in forms with mandatory file questions and requireCompletion = true

marta- opened the pull request:
* See https://phenotips.atlassian.net/browse/CARDS-2587 for bug replication and **testing** guide.

* **Regression testing**: session expiry functionality may have been impacted and needs to be checked

marta- committed:
CARDS-2587: Malfunction in forms with mandatory file questions and requireCompletion = true

marta- committed:
CARDS-2587: Malfunction in forms with mandatory file questions and requireCompletion = true

[refactoring] Simplified the way we check if a save is trigerred automatically

sashaandjic commented:
The red box should disappear and the save button should not be disabled once the file was added.
![image](https://github.com/user-attachments/assets/2b5ff9b2-9c7e-4dc2-867e-1b8ac073ce2c)

Expected behavior is below:
Missing question is in the red box and `Save` button is disabled ( note the button lablel should display Save not Saved)
![image](https://github.com/user-attachments/assets/9e0e661f-ffce-47db-bc3a-a3b0dfb5805c)

Once value is added, the red box disappear and the `Save` button is enabled:
![image](https://github.com/user-attachments/assets/d3cadc0d-bbee-43d2-95eb-2ed4af9a032c)

marta- committed:
CARDS-2587: Malfunction in forms with mandatory file questions and requireCompletion = true

Bugfix: ensure file questions previously marked as incomplete in the UI get cleared once files are uploaded

sashaandjic reviewed (approved):
tested

marta- committed:
CARDS-2587: Malfunction in forms with mandatory file questions and requireCompletion = true

Added a comment to explain a function that was moved to a new location in the file

marta- committed:
CARDS-2587: Malfunction in forms with mandatory file questions and requireCompletion = true

Better function name

veronikaslc commented:
I've tested and looked at code and don't see any problem with this PR.
